romancal/multiband_catalog/multiband_catalog_step.py

```diff
@@ -111,89 +141,89 @@ def process(self, library):
-            with library:
-                for model in library:
-                    mask = (
-                        ~np.isfinite(model.data)
-                        | ~np.isfinite(model.err)
-                        | (model.err <= 0)
-                    )
-
-                    filter_name = model.meta.basic.optical_element  # L3
-                    log.info(f"Creating catalog for {filter_name} image")
-                    ref_file = self.get_reference_file(model, "epsf")
-                    self.log.info("Using ePSF reference file: %s", ref_file)
-                    psf_ref_model = datamodels.open(ref_file)
-                    catobj = RomanSourceCatalog(
-                        model,
-                        segment_img,
-                        None,
-                        star_kernel_fwhm,
-                        fit_psf=self.fit_psf,
-                        detection_cat=det_catobj,
-                        mask=mask,
-                        psf_ref_model=psf_ref_model,
-                        cat_type="dr_band",
-                    )
-
-                    filter_name = model.meta.basic.optical_element
-                    cat = add_filter_to_colnames(catobj.catalog, filter_name)
-                    # TODO: what metadata do we want to keep, if any,
-                    # from the filter catalogs?
-                    cat.meta = None  # temporary
-                    # outer join prevents empty table if any columns have
-                    # the same name but different values (e.g., repeated
-                    # filter names)
-                    det_cat = join(det_cat, cat, keys="label", join_type="outer")
-                    library.shelve(model, modify=False)
-
-            # put the resulting catalog in the model
-            source_catalog_model.source_catalog = det_cat
-
-        # always save the segmentation image
-        output_filename = (
-            self.output_file
-            if self.output_file is not None
-            else source_catalog_model.meta.filename
+            msg = "Cannot create source catalog. No sources were detected."
+            return save_empty_results(self, det_img.shape, cat_model, msg=msg)
+
+        segment_img.detection_image = det_img
+
+        # Define the detection image model
+        det_model = datamodels.MosaicModel()
+        det_model.data = det_img
+        det_model.err = det_err
+
+        # TODO: this is a temporary solution to get model attributes
+        # currently needed in RomanSourceCatalog
+        det_model.weight = example_model.weight
+        det_model.meta = example_model.meta
+
+        # The stellar FWHM is needed to define the kernel used for
+        # the DAOStarFinder sharpness and roundness properties.
+        # TODO: measure on a secondary detection image with minimal
+        # smoothing?; use the same detection image for basic shape
+        # measurements?
+        star_kernel_fwhm = np.min(self.kernel_fwhms)
```

Comment:
Can we grab this from the L3 PSF model?

Reply:
`DAOStarFinder` requires the FWHM as a float (one cannot input a kernel), but we could measure that from the L3 PSF model. 

We should also change this:
https://github.com/spacetelescope/romancal/pull/1822/files/37202edafa6b37163a093cdaa204a18a7a9db3f3#diff-e0dd4868844daac4a496756b08b6d773bc30a1158d29f37ee1c5b22e28c4ecdfR108-R112

to use the L3 PSF model as the kernel (plus another "galaxy" kernel).  I can change the logic there to pass in kernels instead of defining (Gaussian) FWHMs.

But both of those issues are out of scope for this PR.

Reply:
Yes, agreed---out of scope, and the approach I would take here would be to grab the L3 PSF and measure an equivalent Gaussian size, rather than to derive something directly from the images (i.e., we know what the PSF is and shouldn't need to figure it out from the images).

Reply:
Agreed.